File: domain/src/main/java/io/prizy/domain/contest/mapper/CreateContestMapper.java
package io.prizy.domain.contest.mapper;


import io.prizy.domain.contest.command.CreateContest;
import io.prizy.domain.contest.model.Contest;
import lombok.experimental.UtilityClass;

@UtilityClass
public class CreateContestMapper {

  public Contest map(CreateContest create) {
    return Contest.builder()
      .name(create.name())
      .description(create.description())
      .category(create.category())
      .fromDate(create.fromDate())
      .toDate(create.toDate())
      .assetIds(create.assetIds())
      .coverAssetId(create.coverAssetId())
      .packs(create.packs().stream().map(CreatePackMapper::map).toList())
      .cost(create.cost())
      .facebookPage(create.facebookPage())
      .instagramPage(create.instagramPage())
      .website(create.website())
      .newsletterSubscription(create.newsletterSubscription())
      .adultOnly(create.adultOnly())
      .merchant(create.merchant())
      .boardId(create.boardId())
      .build();
  }

}

File: public-api/src/main/kotlin/io/prizy/publicapi/port/contest/mapper/CreateContestMapper.kt
package io.prizy.publicapi.port.contest.mapper

import io.prizy.domain.contest.command.CreateContest
import io.prizy.domain.contest.model.ContestCategory
import io.prizy.publicapi.port.contest.graphql.dto.ContestCategoryDto
import io.prizy.publicapi.port.contest.graphql.dto.CreateContestDto
import java.util.Optional

/**
 *  @author Nidhal Dogga
 *  @created 4/24/2022 8:47 PM
 */

object CreateContestMapper {

  fun map(dto: CreateContestDto): CreateContest = CreateContest(
    dto.name,
    dto.description,
    map(dto.category),
    dto.fromDate,
    dto.toDate,
    dto.mediaIds,
    dto.coverMediaId,
    dto.packs.map(CreatePackMapper::map),
    dto.cost,
    Optional.ofNullable(dto.facebookPage),
    Optional.ofNullable(dto.instagramPage),
    Optional.ofNullable(dto.website),
    dto.newsletterSubscription,
    dto.adultOnly,
    MerchantDtoMapper.map(dto.merchant),
    dto.boardId,
  )

  fun map(dto: ContestCategoryDto): ContestCategory = ContestCategory.valueOf(dto.name)
}
